src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/ExtendedQueryTagDataRowsBuilder.cs

```diff
@@ -95,25 +101,25 @@ private static void AddPersonNameRow(DicomDataset instance, List<InsertPersonNam
 
         private static void AddDateTimeRow(DicomDataset instance, List<InsertDateTimeExtendedQueryTagTableTypeV1Row> dateTimeRows, QueryTag queryTag)
         {
-            DateTime? dateVal = DataTimeReaders.TryGetValue(
-                             queryTag.VR,
-                             out Func<DicomDataset, DicomTag, DicomVR, DateTime?> reader) ? reader.Invoke(instance, queryTag.Tag, queryTag.VR) : null;
+            DateTimeOffset? dateVal = DataTimeReaders.TryGetValue(
+                                         queryTag.VR,
+                                         out Func<DicomDataset, DicomTag, DicomVR, DateTimeOffset?> reader) ? reader.Invoke(instance, queryTag.Tag, queryTag.VR) : null;
 
             if (dateVal.HasValue)
             {
-                dateTimeRows.Add(new InsertDateTimeExtendedQueryTagTableTypeV1Row(queryTag.ExtendedQueryTagStoreEntry.Key, dateVal.Value, (byte)queryTag.Level));
+                dateTimeRows.Add(new InsertDateTimeExtendedQueryTagTableTypeV1Row(queryTag.ExtendedQueryTagStoreEntry.Key, dateVal.Value.DateTime, (byte)queryTag.Level));
             }
         }
 
         private static void AddDateTimeWithUtcRow(DicomDataset instance, List<InsertDateTimeExtendedQueryTagTableTypeV2Row> dateTimeRows, QueryTag queryTag)
         {
-            DateTime? dateVal = DataTimeReaders.TryGetValue(
+            DateTimeOffset? dateVal = DataTimeReaders.TryGetValue(
                              queryTag.VR,
-                             out Func<DicomDataset, DicomTag, DicomVR, DateTime?> reader) ? reader.Invoke(instance, queryTag.Tag, queryTag.VR) : null;
+                             out Func<DicomDataset, DicomTag, DicomVR, DateTimeOffset?> reader) ? reader.Invoke(instance, queryTag.Tag, queryTag.VR) : null;
 
             if (dateVal.HasValue)
             {
-                dateTimeRows.Add(new InsertDateTimeExtendedQueryTagTableTypeV2Row(queryTag.ExtendedQueryTagStoreEntry.Key, dateVal.Value, null, (byte)queryTag.Level));
+                dateTimeRows.Add(new InsertDateTimeExtendedQueryTagTableTypeV2Row(queryTag.ExtendedQueryTagStoreEntry.Key, dateVal.Value.DateTime, queryTag.VR == DicomVR.DT ? dateVal.Value.UtcDateTime : null, (byte)queryTag.Level));
```

Comment:
dateVal.Value.UtcDateTime does not seem correct. It will use the server timezone and convert to UTC.

Reply:
Updated.